repository/src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceFacadeFactoryBean.java

```diff
@@ -351,59 +360,104 @@ else if (isDefined(config.getTruststore()))
+                .map(Builder::build)
+                .orElseThrow(() -> new IllegalStateException("Failed to create ClientRegistration."));
+        }
+
+        private OIDCProviderMetadata validateDiscoveryDocument(OIDCProviderMetadata metadata)
+        {
+            validateOIDCEndpoint(metadata.getTokenEndpointURI(), "Token");
+            validateOIDCEndpoint(metadata.getAuthorizationEndpointURI(), "Authorization");
+            validateOIDCEndpoint(metadata.getUserInfoEndpointURI(), "User Info");
+            validateOIDCEndpoint(metadata.getJWKSetURI(), "JWK Set");
+
+            if (metadata.getIssuer() != null)
+            {
+                try
+                {
+                    URI metadataIssuerURI = new URI(metadata.getIssuer().getValue());
+                    validateOIDCEndpoint(metadataIssuerURI, "Issuer");
+                    if (StringUtils.isNotBlank(config.getIssuerUrl()) &&
+                        !metadataIssuerURI.equals(URI.create(config.getIssuerUrl())))
+                    {
+                        throw new IdentityServiceException("Failed to create ClientRegistration. "
+                            + "The Issuer value from the OIDC Discovery does not align with the provided Issuer. Expected `%s` but found `%s`"
+                            .formatted(config.getIssuerUrl(), metadata.getIssuer().getValue()));
+                    }
+                }
+                catch (URISyntaxException e)
+                {
+                    throw new IdentityServiceException("The Issuer value could not be parsed as a URI reference.", e);
+                }
+            }
+            else
+            {
+                throw new IdentityServiceException("The Issuer cannot be null.");
+            }
+
+            return metadata;
+        }
+
+        private void validateOIDCEndpoint(URI value, String endpointName)
+        {
+            if (value == null || value.toASCIIString().isBlank())
+            {
+                throw new IdentityServiceException("The `%s` Endpoint cannot be empty.".formatted(endpointName));
+            }
         }
 
         private ClientRegistration.Builder createBuilder(OIDCProviderMetadata metadata)
         {
             final String authUri = Optional.of(metadata)
-                                           .map(OIDCProviderMetadata::getAuthorizationEndpointURI)
-                                           .map(URI::toASCIIString)
-                                           .orElse(null);
+                .map(OIDCProviderMetadata::getAuthorizationEndpointURI)
+                .map(URI::toASCIIString)
+                .orElse(null);
 
             final String issuerUri = Optional.of(metadata)
-                    .map(OIDCProviderMetadata::getIssuer)
-                    .map(Issuer::getValue)
-                    .orElseGet(config::getIssuerUrl);
+                .map(OIDCProviderMetadata::getIssuer)
+                .map(Issuer::getValue)
+                .orElse(null);

```

Comment:
I'm assuming this default-change is wanted as we only want to trust the Issuer Url recovered via the `OIDCProviderMetadata` calls?

Reply:
You are right. It's better to trust the Issuer URL from the configuration. Anyway, before that piece of code we validate the OIDC Discovery document endpoints because the Issuer URL is required.